File: java/problem-34.java
public class Problem34 {

	public static int factorial(int n) {
		int returnVal = 1;
		for (int i = 1; i <= n; i++) {
			returnVal *= i;
		}
		return returnVal;
	}

	public static void main(String[] args) {
		int[] facArr = {factorial(0), factorial(1), factorial(2), factorial(3), factorial(4), factorial(5), factorial(6), factorial(7), factorial(8), factorial(9)};
		int facSums = 0;
		int currSum = 0;
		String currNum;
		for (int i = 10; i < 2540160; i++) {
			currSum = 0;
			currNum = "" + i;
			for (int j = 0; j < currNum.length(); j++) {
				currSum += facArr[Integer.parseInt("" + currNum.charAt(j))];
			}
			if (currSum == i) {
				facSums += i;
			}
		}
		System.out.println("The sum of digit factorial numbers is: " + facSums);
	}

}

File: python/problem-34.py
### Problem 34 - Digit Factorials
###-----------------------------------------------------------------------------------------
### 145 is a curious number, as 1! + 4! + 5! = 1 + 24 + 120 = 145.
### Find the sum of all numbers which are equal to the sum of the factorial of their digits.
### Note: as 1! = 1 and 2! = 2 are not sums they are not included.

### Solution

# 7*9! = 2540160 which is less than 9999999 so we will set this as the upper limit

factorials = {'0': 1, '1': 1, '2': 2, '3': 6, '4': 24, '5': 120, '6': 720, '7': 5040, '8': 40320, '9': 362880}
total_sum = 0

for i in range(3, 2540161):
	factorial_sum = 0
	for j in str(i):
		factorial_sum += factorials[j]
	if i == factorial_sum:
		total_sum += i

print('Total digit factorial sum: ' + str(total_sum))